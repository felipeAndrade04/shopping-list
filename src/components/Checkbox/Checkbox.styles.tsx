import styled from 'styled-components/native';
import { CheckboxProps } from './Checkbox.types';

export const CheckboxContainer = styled.TouchableOpacity<CheckboxProps>`
  width: 20px;
  height: 20px;
  display: flex;
  align-items: center;
  justify-content: center;
  border: 2px solid ${({ theme }) => theme.colors.main};
  border-radius: 4px;
  background-color: ${({ value, theme }) => (value ? theme.colors.main : theme.colors.white)};
`;
