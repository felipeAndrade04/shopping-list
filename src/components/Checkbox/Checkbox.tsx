import { FontAwesome5 } from '@expo/vector-icons';
import * as S from './Checkbox.styles';
import { useState } from 'react';
import { CheckboxProps } from './Checkbox.types';
import { View } from 'react-native';

export function Checkbox({ value, onChange }: CheckboxProps) {
  const [isChecked, setIsChecked] = useState(value);

  function handleChangeValue() {
    setIsChecked((prevState) => {
      onChange(!prevState);
      return !prevState;
    });
  }

  return (
    <S.CheckboxContainer value={isChecked} onPress={handleChangeValue} testID="checkbox">
      {isChecked && (
        <View testID="checkbox-icon">
          <FontAwesome5 name="check" size={12} color="white" />
        </View>
      )}
    </S.CheckboxContainer>
  );
}
