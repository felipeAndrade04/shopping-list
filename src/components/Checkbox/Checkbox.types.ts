/* eslint-disable no-unused-vars */
export interface CheckboxProps {
  value?: boolean;
  onChange?: (value: boolean) => void;
}
