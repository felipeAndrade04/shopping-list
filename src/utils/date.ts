const months = ['jan', 'fev', 'mar', 'abr', 'mai', 'jun', 'jul', 'ago', 'set', 'out', 'nov', 'dez'];

export function formatDate(date: Date) {
  const day = date.getDay();
  const month = date.getMonth();

  return `${day} de ${months[month]}`;
}
