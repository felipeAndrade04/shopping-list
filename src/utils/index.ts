export * from './delay';
export * from './date';
