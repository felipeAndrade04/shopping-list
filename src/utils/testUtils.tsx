import React from 'react';
import { authReducer } from '@app/store';
import { ReactNode } from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components/native';
import * as theme from '@app/theme';
import { configureStore } from '@reduxjs/toolkit';

let store = configureStore({
  reducer: {
    auth: authReducer,
  },
});

export const wrapper = ({ children }: { children: ReactNode }) => (
  <ThemeProvider theme={theme}>
    <Provider store={store}>{children}</Provider>
  </ThemeProvider>
);

export function resetStore() {
  store = configureStore({
    reducer: {
      auth: authReducer,
    },
  });
}
