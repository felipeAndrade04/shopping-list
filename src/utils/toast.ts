import EventManager from './EvenManager';

export type ToastType = 'success' | 'info' | 'error';

export interface Toast {
  id?: string;
  type: ToastType;
  text: string;
  duration?: number;
}

export const toastEventManager = new EventManager();

export default function toast({ type, text, duration }: Toast) {
  toastEventManager.emit<Toast>('addtoast', {
    id: new Date().getTime().toString(),
    type,
    text,
    duration,
  });
}
