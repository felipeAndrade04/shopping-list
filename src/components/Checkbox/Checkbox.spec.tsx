import { fireEvent, render } from '@testing-library/react-native';
import { Checkbox } from './Checkbox';
import { wrapper } from '@app/utils/testUtils';

describe('<Checkbox />', () => {
  it('Should be render correctly', () => {
    const tree = render(<Checkbox />, { wrapper }).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should be render checked checkbox', () => {
    const { getByTestId } = render(<Checkbox value={true} />, { wrapper });

    expect(getByTestId('checkbox-icon')).toBeTruthy();
  });

  it('Should be render unchecked checkbox', () => {
    const { queryByTestId } = render(<Checkbox value={false} />, { wrapper });

    expect(queryByTestId('checkbox-icon')).toBeFalsy();
  });

  it('Should toggle checkbox', () => {
    let value = true;

    const { getByTestId } = render(<Checkbox value={value} onChange={() => (value = !value)} />, {
      wrapper,
    });

    expect(value).toBeTruthy();

    const checkbox = getByTestId('checkbox');
    fireEvent.press(checkbox);

    expect(value).toBeFalsy();
  });
});
