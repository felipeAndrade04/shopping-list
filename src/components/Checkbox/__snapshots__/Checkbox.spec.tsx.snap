// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> Should be render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderColor": "#009Cf7",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 2,
      "display": "flex",
      "height": 20,
      "justifyContent": "center",
      "opacity": 1,
      "width": 20,
    }
  }
  testID="checkbox"
/>
`;
